### DISCUSSION TITLE
Question on `NodeSettings`

### DISCUSSION THREAD

**Discussion by minshao** [aicers/review-web#221] [Q&A] [answered]
In `NodeSettings`:
```Rust
struct NodeSettings {
   ...
  pub sensors: bool,
  pub sensor_list: HashMap<String, bool>,
}
```
I'm not very sure about the usage and the implications of these two fields.
1. `NodeSettings::sensors`: does it implicate whether the node contain a list of sensors?
2. `NodeSettings::sensor_list`: does it hold a list of sensors the node contains? Is the key the name of the sensor? And what is the meaning of the value? 

@sophie-cluml @sehkone, any ideas? Appreciate the help~

**Comment by sophie-cluml** [comment, marked as answer]
Hello, 
- `NodeSettings::sensor_list` is the configuration settings field for Hog. In Hog, it is called `active_sources`. The example of keys are like https://github.com/aicers/hog/blob/main/README.md?plain=1#L39. The value of `sensor_list` map is decided by the user in the UI. If it is `true`, it means user selected that one.
- `NodeSettings::sensors` means if user clicked the checkbox "sensor" in the UI. This is also related to Hog. This does not mean that the node has a sensor module on it.
